synapseclient/client.py

```diff
@@ -972,8 +972,9 @@ def store(self, obj, *, createOrUpdate=True, forceVersion=True, versionLabel=Non
                 properties['dataFileHandleId'] = bundle['entity']['dataFileHandleId']
 
             # update the file_handle metadata if the FileEntity's FileHandle id has changed
-            if '_file_handle' in local_state \
```

Comment:
bug was caused by `'_file_handle'` in `local_state` being initialized elsewhere with `{'id':None , ...}`, which causes `properties['dataFileHandleId'] != local_state['_file_handle'].get('id', None)` to evaluate as `True` when an existing FileHandle was already found on Synapse causing a value to be set for `properties['dataFileHandleId']`